### PR TITLE
Filtering apps

### PR DESCRIPTION
**Pull request by fatima-rizvi**
# Pull Request Template

## Description

I set up state and retrieved data from the database to display on SearchPage. I moved the state management for filterValues to SearchPage so that when SearchInput updates the filterValues, SearchPage has access to the new filters and can pass the information to CompactApp. CompactApp now contains two conditons built through nested if statements that filter out applications based on filterValues (name and status).

Also created a new file called dummySearchData to use when desired in the search component instead of retrieving information from the backend. This is useful (at times) because the current applications in the backend all have a status of "not reviewed" and share similar names, making it hard to test SearchInput.

**Note:** Although the functions to accept and reject applications have been defined, they don't currently do either of those things. Also, due to issues logging in locally, I was unable to use the react queries (as they require authorization). Future work that needs to be done includes using the react queries to retrieve the applications that are specific to the user logged in (meaning each organization should only be able to access applications submitted to themselves) and giving the handleAccept and handleReject functions in CompactApp the ability to toggle the individual application's status.

## Trello link (and other relevent links)

[Trello Card](https://trello.com/c/2c8MCw6B)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Also

- [x] Use small commits with clear, relevant titles and descriptions

- [x] Scope the pull request to a single relevant feature

- [x] Remove print statements and commented out code

- [ ] Review by two team-members before merging

### CODE REVIEW

File: src/components/pages/Search/CompactApp.js [hunk 1]
```diff
@@ -18,22 +16,39 @@ function CompactApp({ app }) {
...
+    ) {
+      return null;
+    }
+  }
+
+  // Filters out apps based on status
+  if (filterValues.status !== 'all') {
+    if (filterValues.status !== '') {
+      if (filterValues.status !== null && app.status !== filterValues.status) {
+        return null;
```
Comment (buchananta) [inline src/components/pages/Search/CompactApp.js:47]: Is there a reason we're nesting if statements and not just doing something like
```
if (filterValues.status !== 'all'
 && filterValues.status !== ''
 && filterValues.status !== null
 && app.status !== filterValuse.status ) {
    return null;
}
```
I'm sure it works fine either way, just looks weird to me

**Review by buchananta — APPROVED** [review]
Looks great! I'd personally use the boolean `&&` rather than nesting multiple if statements. Works the same either way!

**Review by rodhent — APPROVED** [review]
awesome! thanks

File: src/components/pages/Search/CompactApp.js (hunk 1 above)
Comment (fatima-rizvi) [reply src/components/pages/Search/CompactApp.js:47]: I think that was just for keeping the conditions straight in my head, but I can go back and fix that in a later PR. Thank you!

**Pull request merged**